src/components/DelegationDetailView/DelegationDetailView.tsx

```diff
@@ -62,93 +61,73 @@ class DelegationDetailView extends React.Component<Props, State> {
     return node
   }
 
-  constructor(props: Props) {
-    super(props)
-    this.state = {
-      rootNode: null,
-    }
-  }
-
-  public componentDidMount(): void {
-    const { id } = this.props
-
-    DelegationDetailView.getNode(id)
+  const delegationTreeForester = (): void => {
+    getNode(delegationId)
       .then(async (delegationNode: sdk.DelegationNode) => {
         const treeNode: DelegationsTreeNode = {
           childNodes: [],
           delegation: delegationNode,
-        } as DelegationsTreeNode
-
-        const rootNode: State['rootNode'] = await DelegationDetailView.resolveRootNode(
-          treeNode
+        }
+        const rootNodeFind = await DelegationsService.findRootNode(
+          treeNode.delegation.id
         )
-        const parentTreeNode: DelegationsTreeNode = await DelegationsService.resolveParent(
+        setRootNode(rootNodeFind)
+
+        const delegationTreeNode = await DelegationsService.resolveParent(
           treeNode
         )
-
-        this.setState({
-          delegationsTreeNode: parentTreeNode || treeNode,
-          rootNode,
-        })
+        setDelegationsTreeNode(delegationTreeNode)
       })
       .catch(error => {
         console.log('error', error)
       })
   }
 
-  public render(): JSX.Element {
-    const {
-      viewType,
-      editable,
-      id,
-      isPCR,
-      focusedNodeAlias,
-      selectedIdentity,
-    } = this.props
-    const { delegationsTreeNode, rootNode } = this.state
-
-    return (
-      <section className="DelegationDetailView">
-        <h1>{isPCR ? 'PCR view' : 'Delegation tree'}</h1>
-        <div className="delegationNodeContainer">
-          {delegationsTreeNode && (
-            <>
-              {rootNode && (
-                <h2>
-                  <span>CType: </span>
-                  <CTypePresentation
-                    cTypeHash={rootNode.cTypeHash}
-                    interactive
-                    linked
-                    inline
-                  />
-                </h2>
-              )}
-              {!rootNode && <h2>No CType!</h2>}
-              <br />
-              <div className="delegationNodeScrollContainer">
-                <DelegationNode
-                  key={delegationsTreeNode.delegation.id}
-                  node={delegationsTreeNode}
-                  selectedIdentity={selectedIdentity}
-                  focusedNodeId={id}
-                  focusedNodeAlias={focusedNodeAlias}
-                  editable={editable}
-                  viewType={viewType}
+  useEffect(() => {
+    delegationTreeForester()
+  }, [delegationId, rootNode, delegationsTreeNode])
```

Comment:
Checking for `rootNode, delegationsTreeNode` will make a loop, since they are state values and you are updating the state inside the effect function